lib/fetch/index.js

```diff
@@ -122,130 +124,170 @@ class Fetch extends EE {
   }
 }
 
+const subscribed = (name) => {
+  if (channels[name].hasSubscribers) {
+    channels[name].publish(context)
+  }
+}
+const callAsyncStartEnd = () => {
+  subscribed('asyncStart')
+  subscribed('asyncEnd')
+}
+
 // https://fetch.spec.whatwg.org/#fetch-method
-function fetch (input, init = undefined) {
+function fetch (input, init = {}) {
+  const context = { input, init }
```

Comment:
```suggestion
  const context = { input, init, result: null, error: null }
```

Let's add all 4 props so V8 can optimize this properly

Reply:
added additional props